### PR TITLE
 added mllama doc

### PR DESCRIPTION
**Pull request by Nikil-D-Gr8**
What does this PR do?
As suggested in this issue[ #issue-2947704577](https://github.com/huggingface/transformers/issues/36979#issuecomment-2817025078) - this PR updates the documentation of the CLIP model, which will now be aligned with the standardized format for all the docs.
Worked on mllama, Used AI , so please let me know even if u need a complete rewrite.

- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).

Please let me know if there are any changes to be done, do share references if any for those changes
Documentation: @stevhliu

 -->

### CODE REVIEW

File: docs/source/en/model_doc/mllama.md [hunk 1]
```diff
@@ -14,93 +14,91 @@ rendered properly in your Markdown viewer.
...
-inputs = processor(text=prompt, images=raw_image, return_tensors="pt").to(model.device)
-output = model.generate(**inputs, do_sample=False, max_new_tokens=25)
-print(processor.decode(output[0], skip_special_tokens=True))
-```
+## Notes
 
+- MLlama has special handling for image tokens using `<|image|>` as a placeholder in the text
+- The model's input and output embeddings are not tied, requiring special handling for the `lm_head` layer
+- When training, mask out the `<|image|>` tokens in labels
+- For CUDA index errors during generation, expand the `lm_head`:
```
Comment (stevhliu) [inline docs/source/en/model_doc/mllama.md:93]: Adding a few more notes. You can remove the `\` I added at the end of the code snippet

```suggestion
- Use [`MllamaForConditionalGeneration`] for image-text and text inputs.
- Use [`MllamaForCausalLM`] for generation with text only inputs to avoid unnecessarily loading the vision components.
- Samples can contain multiple images and each sample can have different number of images. In these cases, the processor pads the inputs to the maximum number of images across samples and to a maximum number of tiles within each image.
- Use the `<|image|>` token to indicate where an image should be inserted.

   ```py
   prompt = "<|image|> What does the image show?"
   \```

- MLlama's input and output embeddings are not tied which means the `lm_head` layer has one less token (the `<|image|>` placeholder token) and fails if you want to calculate loss on image tokens or apply logit processors. For training, make sure to mask the `<|image|>` token in `labels` because the model should not be trained on predicting them.
- For CUDA index errors during generation, expand the `lm_head` by one token.
```

File: docs/source/en/model_doc/mllama.md [hunk 2]
```diff
@@ -14,93 +14,91 @@ rendered properly in your Markdown viewer.
...
+
+```python
```
Comment (stevhliu) [inline docs/source/en/model_doc/mllama.md:95]: Indent this code block so it falls under the last list item

File: docs/source/en/model_doc/mllama.md [hunk 3]
```diff
@@ -130,12 +127,4 @@ print(processor.decode(output[0], skip_special_tokens=True))
...
 
-## MllamaForCausalLM
 
-[[autodoc]] MllamaForCausalLM
-    - forward
-
-## MllamaVisionModel
-
-[[autodoc]] MllamaVisionModel
-    - forward
```
Comment (stevhliu) [inline docs/source/en/model_doc/mllama.md:141]: Add these docstrings back!

**Review by stevhliu — COMMENTED** [review]
Awesome, thanks for your contribution!

Let's also add an example with the `AttentionMaskVisualizer`

**Comment by Nikil-D-Gr8** [comment]
@stevhliu Thanks a lot for the help mate,  used ai a bit here and there thinking it can do a better job, looks like it just gave u more work,  new to all this , will keep it organic from now,thanks, let me know if i can make any more edits tho

File: docs/source/en/model_doc/mllama.md [hunk 4]
```diff
@@ -14,93 +14,128 @@ rendered properly in your Markdown viewer.
...
+    ```
+
+<div class="flex justify-center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/model_doc/mllama_architecture.png"/>
+</div>
 
-model_id = "meta-llama/Llama-3.2-11B-Vision"
-model = MllamaForConditionalGeneration.from_pretrained(model_id, device_map="auto", torch_dtype=torch.bfloat16)
-processor = AutoProcessor.from_pretrained(model_id)
+## Notes
```
Comment (stevhliu) [inline docs/source/en/model_doc/mllama.md:118]: You can remove all this since the `## Notes` under it are the same

File: docs/source/en/model_doc/mllama.md [hunk 5]
```diff
@@ -14,93 +14,110 @@ rendered properly in your Markdown viewer.
...
-import torch
-from PIL import Image
-from transformers import MllamaForConditionalGeneration, AutoProcessor
+</hfoption>
+</hfoptions>
 
-model_id = "meta-llama/Llama-3.2-11B-Vision"
-model = MllamaForConditionalGeneration.from_pretrained(model_id, device_map="auto", torch_dtype=torch.bfloat16)
-processor = AutoProcessor.from_pretrained(model_id)
+<div class="flex justify-center">
```
Comment (stevhliu) [inline docs/source/en/model_doc/mllama.md:101]: We can remove this image and replace it with the quantization example in this [comment](https://github.com/huggingface/transformers/pull/37647#discussion_r2054412777).

```py
import torch
from transformers import BitsAndBytesConfig, MllamaForConditionalGeneration, AutoProcessor

bnb_config = BitsAndBytesConfig(
    load_in_4bit=True,
    bnb_4bit_use_double_quant=True,
    bnb_4bit_quant_type="nf4",
    bnb_4bit_compute_dtype=torch.bfloat16
)
model = MllamaForConditionalGeneration.from_pretrained(
    "meta-llama/Llama-3.2-11B-Vision-Instruct",
    device_map="auto", 
    torch_dtype=torch.bfloat16,
    attn_implementation="sdpa",
    quantization_config=bnb_config
)
processor = AutoProcessor.from_pretrained("meta-llama/Llama-3.2-11B-Vision-Instruct")

messages = [
    [
        {
            "role": "user", 
            "content": [
                {"type": "image", "url": "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg"},
                {"type": "text", "text": "What does the image show?"}
            ]
        }
    ],
]
inputs = processor.apply_chat_template(
    messages,
    add_generation_prompt=True,
    tokenize=True,
    return_dict=True,
    return_tensors="pt"
).to("cuda")
output = model.generate(**inputs, max_new_tokens=25)
print(processor.decode(output[0]))
```

File: docs/source/en/model_doc/mllama.md [hunk 6]
```diff
@@ -14,93 +14,110 @@ rendered properly in your Markdown viewer.
...
 
-## Usage Example
+</hfoption>
+<hfoption id="AutoModel">
 
-#### Instruct model
 ```python
 import torch
-from transformers import MllamaForConditionalGeneration, AutoProcessor
+from transformers import BitsAndBytesConfig, MllamaForConditionalGeneration, AutoProcessor
```
Comment (stevhliu) [inline docs/source/en/model_doc/mllama.md:66]:
```suggestion
from transformers import MllamaForConditionalGeneration, AutoProcessor
```

File: docs/source/en/model_doc/mllama.md (hunk 2 above)
Comment (stevhliu) [reply docs/source/en/model_doc/mllama.md:114]: Remember to indent here!

**Comment by Nikil-D-Gr8** [comment]
is it fine now?

File: docs/source/en/model_doc/mllama.md (hunk 6 above)
Comment (stevhliu) [reply docs/source/en/model_doc/mllama.md:66]: The `AutoModel` example shouldn't show quantization usage, so it was fine the way it was before. I was just removing `BitsAndBytesConfig` from the import

File: docs/source/en/model_doc/mllama.md (hunk 5 above)
Comment (stevhliu) [reply docs/source/en/model_doc/mllama.md:108]: Separate from the `AutoModel` example and outside of the `<hfoption>` block, you should have a separate code example for quantization as shown in the code snippet above.

The image hasn't been removed yet

File: docs/source/en/model_doc/mllama.md (hunk 2 above)
Comment (stevhliu) [reply docs/source/en/model_doc/mllama.md:121]: Still not indented

File: docs/source/en/model_doc/mllama.md [hunk 7]
```diff
@@ -125,17 +148,21 @@ print(processor.decode(output[0], skip_special_tokens=True))
...
 
-## MllamaVisionModel
 
-[[autodoc]] MllamaVisionModel
-    - forward
+
+
+
+
+
```
Comment (stevhliu) [inline docs/source/en/model_doc/mllama.md:168]: No need to add all these extra lines at the end either
